'initial' contribution (#13)
swcarpentry/python-intermediate: _episodes/05-APIs.md

---
title: Remote Data and APIs
teaching: 40
exercises: 15
questions:

> - "What is an API?"
> - "How get I automatically retrieve data from an API?"

Reviewer: ```suggestion
- "What is an API?"
- "How to retrieve data from an API?"
```
_Automatically_ is superfluous here. Automation is why one writes a program after all

objectives:
- "Download data from an API."
keypoints:
- "TODO"

> ---

Reviewer: Selecting a good API for this task is going to be a challenge.
It must be _public_, _easy to grasp_ to not distract from the main matter, _have a high availability_ so it does not get impeded by a whole room of people suddenly experimenting on it. You can not count on an API that is self-hosted due to the effort that comes with it.

Consider that for the API all these requests may appear to come from the same domain and it might trigger DDOS protection if something unexpected happens (I forgot to terminate a loop once…)

Reviewer: Maybe this topic is too advanced for an intermediate course?
